### DISCUSSION TITLE
PnP People search Shows AAD disabled account in Search result as well as in Hover Card. Is their any way to exclude those users

### DISCUSSION THREAD

**Discussion by Ami99049** [microsoft-search/pnp-modern-search#3224] [Q&A] [answered]
PnP People search Shows AAD disabled account in Search result as well as in Hover Card. Is their any way to exclude those users

**Comment by kasperbolarsen** [comment]
Hi
Have you looked at the options listed here: https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-useful-People-Search/
HidefromAddressList is often the right choice :-)

The User Profile application does not contains a "DisabledInAAD" property by default, so you will have to use one or more of the existing property, or set up a custom property and a custom sync job

**Comment by Ami99049** [comment]
@kasperbolarsen  
Thanks for your reply.
you mean there isn't any OOTB User property (i.e. enabledAccount) to exclude disabled users and We have to create our own custom property and a AAD sync Job.
Also will the custom property exclude the user from People Hover Card Org Chart.

**Comment by kasperbolarsen** [comment]
The closest to an OOTH enabledAccount property IS the HideFromAddressLists property. Is this property is set to 1/true in AAD, we can use it in PnP Modern Search Web parts to exclude those accounts. OOTH Microsoft Search is also using this property to exclude accounts from the "People" vertical.
Some Orgs are not using HideFromAddressLists and they have to spin up a custom sync job in order to hide those disabled accounts , but only in PnP MSWP, as MS Search can't be modified now, perhaps later on.

**Comment by Ami99049** [comment]
![image](https://github.com/microsoft-search/pnp-modern-search/assets/143168322/8fa9d67b-b949-4025-b5b0-28b4754cc0f6)

I am using {searchTerms} HideFromAddressLists:'1' in Query Template but still Disabled accounts are visible. Am I using wrong syntax. CAn you please let us know what exactly i have to use here

**Comment by kasperbolarsen** [comment]
Most likely those accounts do not have HideFromAddressList set to 1/true.
I suggest checking 
1) values in you search index. Use SP Search query tool or similar
2) the values in you User Profiles 
3) the values in your AD , some organizations do not set HideFromAddressLists to true when accounts are disabled

**Comment by Ami99049** [comment]
Thanks @kasperbolarsen 
Let me double check all these things
I also want to know is this approach work for PnP Org Chart too or only for PnP Search Results

**Comment by kasperbolarsen** [comment]
> Thanks @kasperbolarsen 
> Let me double check all these things
> I also want to know is this approach work for PnP Org Chart too or only for PnP Search Results

What is PnP Org Chart, do you have a link?

**Comment by Ami99049** [comment]
When the persona setting for PNP people search is ON and we hover over a user profile it's show the 
![Screenshot_20230825-195114](https://github.com/microsoft-search/pnp-modern-search/assets/143168322/e11174ac-33c8-4c5b-bbc1-9b9f338ae986)

When you click on bottom arrow it's show few more people ..i.e. organization hierarchy of the user. In that hover card some of the users are disabled but they are still visible.

**Comment by kasperbolarsen** [comment]
Good question. As I understand the setup the data in that component is provided by MS Graph, and I would expect that it will exclude accounts having HideFromAddressList=1, but I could be wrong. 
Could you find an Account in your environment that has HideFromAddressList=1 and check if it is shown in the Org chart ?

**Comment by kasperbolarsen** [comment, marked as answer]
The correct query is "SPS-HideFromAddressLists"=1 for getting the disabled accounts, sorry. I have just checked if account with "SPS-HideFromAddressLists"=1 show up in the Org Char: they do not :-)

**Comment by Ami99049** [comment]
Thanks @kasperbolarsen 
Let me double check all these at my and and get back to you.

**Reply by gretchunkim** [reply]
@Ami99049  @kasperbolarsen  
Also facing the same issue, and using "SPS-HideFromAddressLists"=1  or "SPS-HideFromAddressLists"=0 doesn't fix the issue... 
The underlying issue I believe is within AAD/On-prem AD. Will post more once I find more info.

**Comment by adumith** [comment]
This property "SPS-HideFromAddressLists"=0 worked for me.

I no longer see the disabled accounts, but I still see other accounts such as conference rooms, which have NO department assigned. I wonder how I can use the department property to show only those users that have a department assigned or I can take a look to another property for it?

Thanks in advance,
